plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/AbstractFilesTable.java

```diff
@@ -114,7 +112,21 @@ protected ConnectorTableMetadata tableMetadata(SchemaTableName tableName, TypeMa
                         .build());
     }
 
-    protected void addRow(PageListBuilder pagesBuilder, Map<Integer, Type> idToTypeMapping, StructLike structLike)
+    private List<Page> buildPages()
```

Comment:
Does this belong to the commit which introduced `AbstractFilesTable` class?

Reply:
yes @findepi this was originally added with the `AbstractFilesTable` class, and then refactored to keep it private.
is there any issue with this?

Reply:
> Does this belong to the commit which introduced `AbstractFilesTable` class?

@findepi no this commit was not belonged to this change (adding `buildPages()`).
Firstly `AbstractFilesTable` being added and then `buildPages()` being added.

Reply:
@findepi can I request, if we can squash the commit for this PR and use.
The `AbstractFilesTable` was added after the initial round of functional changes.
So, it seems to be bit more work to extract out changes and have 2 different commits

Hoping be from next time it will be better.

1. refactor existing code to add `AbstractFilesTable`
2. add `AllFiles` table + test cases

thanks a lot!

cc @alexjo2144